WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java

```diff
@@ -94,60 +98,95 @@ public void onCreate(Bundle savedInstanceState) {
-
-        //set up the viewpager and adapter for lateral navigation
-        mViewPager = (WPViewPager) findViewById(R.id.viewpager);
-        mViewPager.setPageTransformer(false,
-                new WPViewPagerTransformer(WPViewPagerTransformer.TransformType.SLIDE_OVER));
-
         NotesAdapter.FILTERS filter = NotesAdapter.FILTERS.FILTER_ALL;
         if (getIntent().hasExtra(NotificationsListFragment.NOTE_CURRENT_LIST_FILTER_EXTRA)) {
             filter = (NotesAdapter.FILTERS) getIntent().getSerializableExtra(NotificationsListFragment.NOTE_CURRENT_LIST_FILTER_EXTRA);
         }
+
         mAdapter = buildNoteListAdapterAndSetPosition(note, filter);
 
+        resetOnPageChangeListener();
+
         //set title
         setActionBarTitleForNote(note);
         markNoteAsRead(note);
 
-        mViewPager.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
-            @Override
-            public void onPageSelected(int position) {
-                AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATION_SWIPE_PAGE_CHANGED);
-                AppPrefs.setNotificationsSwipeToNavigateShown(true);
-                //change the action bar title for the current note
-                Note currentNote = mAdapter.getNoteAtPosition(position);
-                if (currentNote != null) {
-                    setActionBarTitleForNote(currentNote);
-                    markNoteAsRead(currentNote);
+        // If `note.getTimestamp()` is not the most recent seen note, the server will discard the value.
+        NotificationsActions.updateSeenTimestamp(note);
+
+        // analytics tracking
+        Map<String, String> properties = new HashMap<>();
+        properties.put("notification_type", note.getType());
+        AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATIONS_OPENED_NOTIFICATION_DETAILS, properties);
+
+        setProgressVisible(false);
+    }
+
+    private void resetOnPageChangeListener() {
+        if (mOnPageChangeListener != null) {
+            mViewPager.removeOnPageChangeListener(mOnPageChangeListener);
+        } else {
+            mOnPageChangeListener = new ViewPager.OnPageChangeListener() {
+                @Override
+                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
                 }
-            }
-        });
 
-        // Hide the keyboard, unless we arrived here from the 'Reply' action in a push notification
-        if (!getIntent().getBooleanExtra(NotificationsListFragment.NOTE_INSTANT_REPLY_EXTRA, false)) {
-            getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_HIDDEN);
+                @Override
+                public void onPageSelected(int position) {
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATION_SWIPE_PAGE_CHANGED);
+                    AppPrefs.setNotificationsSwipeToNavigateShown(true);
```

Comment:
Arguably not directly related to this PR, but good call nevertheless, addressed here c96d896 :)